ext/XS-Typemap/Typemap.xs: build-time warning with clang-14

When compiling perl with clang (here, version 14), we're emitting a build-time warning that I had previously overlooked.
```
$ uname -mrs
Linux 6.2.0-33-generic x86_64

$ git describe
v5.39.3-9-g1c5a7c530e

$ ./perl -Ilib -V:config_args
config_args='-des -Dusedevel -Dcc=clang-14';

$ parse-build-warnings 1c5a7c530e.linux.unthreaded.clang-14.maketp.output.txt.gz
File:  1c5a7c530e.linux.unthreaded.clang-14.maketp.output.txt.gz

[
  # 3 build-time error in code maintained upstream on CPAN;
  # not relevant here
  ...
  {
    char   => 6,
    group  => "Wunused-but-set-variable",
    line   => 1775,
    source => "Typemap.c",
    text   => "variable 'dummy' set but not used",
  },
]
```
I've been observing this in clang builds on both FreeBSD-13 and Ubuntu Linux 22.04 LTS for several months, on both threaded and unthreaded builds -- but I long thought that `Typemap.c` was maintained upstream on CPAN.  That was wrong.  AFAICT it is C code compiled from `ext/XS-Typemap/Typemap.xs`, which is blead's responsibility.

In the output of `make test_prep` I observe:
```
./miniperl -Ilib make_ext.pl lib/auto/XS/Typemap/Typemap.so  MAKE="make" LIBPERL_A=libperl.a LINKTYPE=dynamic
Generating a Unix-style Makefile
Writing Makefile for XS::Typemap
make[1]: Entering directory '/home/jkeenan/gitwork/perl/ext/XS-Typemap'
Running Mkbootstrap for Typemap ()
chmod 644 "Typemap.bs"
/home/jkeenan/gitwork/perl/ext/XS-Typemap/../../miniperl "-I../../lib" -MExtUtils::Command::MM -e 'cp_nonempty' -- Typemap.bs ../../lib/auto/XS/Typemap/Typemap.bs 644
clang-14 -c   -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wall -Werror=pointer-arith -Werror=vla -Wextra -Wno-long-long -Wno-declaration-after-statement -Wc++-compat -Wwrite-strings -O2   -DVERSION=\"0.19\" -DXS_VERSION=\"0.19\" -fPIC "-I../.."   stdio.c
"../../miniperl" "-I../../lib" "../../lib/ExtUtils/xsubpp"  -typemap '/home/jkeenan/gitwork/perl/ext/XS-Typemap/../../lib/ExtUtils/typemap'  Typemap.xs > Typemap.xsc
mv Typemap.xsc Typemap.c
clang-14 -c   -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wall -Werror=pointer-arith -Werror=vla -Wextra -Wno-long-long -Wno-declaration-after-statement -Wc++-compat -Wwrite-strings -O2   -DVERSION=\"0.19\" -DXS_VERSION=\"0.19\" -fPIC "-I../.."   Typemap.c
Typemap.c:1775:6: warning: variable 'dummy' set but not used [-Wunused-but-set-variable]
        int     dummy = 0;
                ^
1 warning generated.
rm -f ../../lib/auto/XS/Typemap/Typemap.so
clang-14  -shared -O2 -L/usr/local/lib -fstack-protector-strong  stdio.o  Typemap.o  -o ../../lib/auto/XS/Typemap/Typemap.so  \
      \
  
chmod 755 ../../lib/auto/XS/Typemap/Typemap.so
make[1]: Leaving directory '/home/jkeenan/gitwork/perl/ext/XS-Typemap'
```
This appears to come from this source code:
```
$ head -n 929 ext/XS-Typemap/Typemap.xs | tail -n 31
## T_ARRAY

# Test passes in an integer array and returns it along with
# the number of elements
# Pass in a dummy value to test offsetting

# Problem is that xsubpp does XSRETURN(1) because we arent
# using PPCODE. This means that only the first element
# is returned. KLUGE this by using CLEANUP to return before the
# end.
# Note: I read this as: The "T_ARRAY" typemap is really rather broken,
#       at least for OUTPUT. That is apart from the general design
#       weaknesses. --Steffen

intArray *
T_ARRAY( dummy, array, ... )
  int dummy = 0;
  intArray * array
 PREINIT:
  U32 size_RETVAL;
 CODE:
  dummy += 0; /* Fix -Wall */
  size_RETVAL = ix_array;
  RETVAL = array;
 OUTPUT:
  RETVAL
 CLEANUP:
  Safefree(array);
  XSRETURN(size_RETVAL);

```
Comments aside, this code was written back in 2001!  Is it possible that modern compilers have obviated the need for some the kludges seen above?

--- a/ext/XS-Typemap/Typemap.pm
+++ b/ext/XS-Typemap/Typemap.pm
@@ -34,7 +34,7 @@ to the test script.
 use parent qw/ Exporter /;
 require XSLoader;
 
-our $VERSION = '0.19';
+our $VERSION = '0.20';
 
 our @EXPORT = (qw/
 	   T_SV

--- a/ext/XS-Typemap/Typemap.xs
+++ b/ext/XS-Typemap/Typemap.xs
@@ -917,7 +917,7 @@ T_ARRAY( dummy, array, ... )
  PREINIT:
   U32 size_RETVAL;
  CODE:
-  dummy += 0; /* Fix -Wall */
+  PERL_UNUSED_VAR(dummy); /* GH 21505 */
   size_RETVAL = ix_array;
   RETVAL = array;
  OUTPUT:

